main.cpp

```diff
@@ -31,6 +31,11 @@
 #pragma warning(disable : 4996)
 #endif
 
+#ifdef __clang__
+#pragma clang diagnostic ignored "-Wswitch-enum"
```

Comment:
For these diagnostics to be relevant, the build system should have an option to build with the strictest possible warning level, or this will not be maintainable.